src/AblyRest.php

```diff
@@ -18,6 +18,16 @@ class AblyRest {
     public $options;
     protected static $libFlavour = '';
 
+    static function ablyAgentHeader()
+    {
+        $sdk_identifier = 'ably-php/'.self::LIB_VERSION;
+        $runtime_identifier = 'php/'.Miscellaneous::getNumeric(phpversion(null));
+        $agent_identifier = $sdk_identifier.' '.$runtime_identifier;
+        if (self::$libFlavour == 'laravel') {
+            $agent_identifier.= ' laravel';
```

Comment:
Is there anyway we can get the version of laravel?

Reply:
We are currently setting it via https://github.com/ably/ably-php-laravel.

https://github.com/ably/ably-php-laravel/blob/3269fa9a51492916b44934eab7d512d6c67a2ae2/src/AblyFactory.php#L35-L40

Reply:
Yes, it's recommended to use that library for `laravel`, so it will be implicitly set.

Reply:
Sorry, I think my question wasn't very clear.

So here, we're setting `ably-php` and `php` in the header with their version numbers, so I'm asking if we can support doing the same for laravel, so we also know what versions of laravel are being used?

For example:

```
ably-php/1.2.3 php/8.1.3 laravel/9.5.1
```

Reply:
I think it's possible to get `laravel` version https://beyondco.de/blog/how-to-check-the-laravel-version-of-your-application. I still need to check, how feasible it is to get it inside laravel app (since it needs an active app instance, **`app()->version()`**). I will probably need help from @owenpearson

Reply:
My PHP isn't great so I'm not sure how much help I would be. My brief googling seems to indicate that we can only do this if we required users to pass in their app instance.

Reply:
Yep, currently, we have it available at two places. https://github.com/ably/ably-php-laravel/blob/3269fa9a51492916b44934eab7d512d6c67a2ae2/src/AblyServiceProvider.php#L33-L40
and 
https://github.com/ably/ably-php-laravel/blob/3269fa9a51492916b44934eab7d512d6c67a2ae2/src/AblyFactory.php#L35-L40

Reply:
Seems it's a global laravel helper as per https://laravel.com/docs/9.x/helpers. So, we should be able to access it directly https://laravel.com/docs/9.x/helpers#method-app.

Reply:
Ok thanks.

So what I think we need to do here is provide an API that allows the caller (e.g. the laravel librray) to provide their own supplementary agent values, rather than hard coding an expectation that it can only be `laravel` with no version.

That way, we can merge this change, and then separately we can update ably-php-laravel to include the laravel version if possible, otherwise just set the agent to `laravel`.

We take a similar approach in ably-java where the caller can set the `agents` client option to a map of extra agents to include (see [here](https://github.com/ably/ably-java/blob/main/lib/src/main/java/io/ably/lib/types/ClientOptions.java#L212-L222)), and then that's set in wrappers like ably-asset-tracking-android (e.g. [here](https://github.com/ably/ably-asset-tracking-android/blob/main/common/src/main/java/com/ably/tracking/common/Ably.kt#L199))

Reply:
Worth mentioning also that [this line in ably-common](https://github.com/ably/ably-common/blob/main/protocol/agents.json#L203) will need to be updated if we're sending laravel versions.

Reply:
Yes, of course we will update that one : )

Reply:
Updated code with `agents` map, responsible for maintaining `agent identifier` and respecive `versions`

Reply:
Nice 👍